### PR TITLE
adding event generation workflow

### PR DESCRIPTION
**Pull request by vsoch**
This is a first PR to start testing the PDF generation workflow. We will likely want to:
 
 - upload a higher definition logo for SORSE
 - decide on some action beyond saving artifacts to do
 - ensure that the rendering is okay
 - and finally, update the version to a release

I'll keep updating the PR if/when things don't work as expected!

### CODE REVIEW

**Comment by vsoch** [comment]
Wow, it worked the first time, amazing! It looks like the events don't have dois, so that isn't rendering. I can try adding an if statement to skip it entire if it's not there.

**Comment by Chilipp** [comment]
Awesome :partying_face: The _Event Website_ is also not working, and we should remove the _Repository_ link, or should the latter point to the markdown file in this repository? This would also be a possibility

**Comment by vsoch** [comment]
okay getting close - I'm going to try and customize the "published" variable so it's not ugly.

**Comment by vsoch** [comment]
Repository link is removed. For event website, I was expecting a `website` variable in the markdown, I'll add an if statement there too.

**Comment by vsoch** [comment]
I'm paused for a second because I'm testing the container to use Python3 so it doesn't break come 2021 :)

**Comment by Chilipp** [comment]
> For event website, I was expecting a website variable in the markdown, I'll add an if statement there too.

coming from jekyll, it would be misleading to add the website to the yaml frontmatter. can't we add it with a `-V` option as suggested in https://github.com/SORSE/sorse.github.io/issues/215#issuecomment-654267603?

**Comment by vsoch** [comment]
Looks like the footer needs more cleaning up and the date is being truncated. I'll  need to debug this in the base action.

**Comment by vsoch** [comment]
@Chilipp yes that's the point of the sorse-variables.yml and sorse-mapping.yml, both are added with -V. It is a general tool so we don't hard code anything like that :)

**Comment by Chilipp** [comment]
don't worry about the date. we will make that shorter now that the time goes in as a separate field

**Comment by Chilipp** [comment]
I'd suggest to just use a format like

```yaml
date: 2020-06-30
```

**Comment by vsoch** [comment]
I think for the calendar if the events have times we would want that to render, so instead I added the publication date to be generated when the PDF is built, because technically that's when it's published. There is also now a check to not re-generate files that already exist (so a pdf is not re-published).

**Comment by vsoch** [comment]
This is the "html-proofer needs a retry" that I mentioned earlier

```
External link https://allejo.io/blog/a-jekyll-toc-in-liquid-only/ failed: got a time out (response code 0)
```
I'm debugging something with the mapping file, will need a few minutes.

**Comment by Chilipp** [comment]
> because technically that's when it's published

not really, it's published when it appears on the website and that is what the `date` key is translated into, right? As discussed in https://github.com/Chilipp/sorse.github.io/pull/3, events now have a `time` and `date` key. You can see the differences in https://23-276193679-gh.circle-artifacts.com/0/SORSE/programme/workshops/event-004/index.html. the `time` is used in the calendar and renders like
image
on the event website, whereas the `date` renders like 
image
and should be the same as the date of the PDF.

**Comment by vsoch** [comment]
okay, I'll change it back to use date then.

**Comment by vsoch** [comment]
@Chilipp the published date is ugly again, but since you have a plan I'll let you take it from. On my action end everything is working as expected, so I'm going to update one last time to include a released version and remove the PR trigger.

**Comment by Chilipp** [comment]
> @Chilipp the published date is ugly again

sure, because https://github.com/Chilipp/sorse.github.io/pull/3 where I changed the dates in the markdown files is not yet merged

**Comment by Chilipp** [comment]
> On my action end everything is working as expected, so I'm going to update one last time to include a released version and remove the PR trigger.

In https://github.com/SORSE/sorse.github.io/issues/215#issue-651040515 I proposed to rather make a PR into an open PR to update the PDFs. That way we can directly add the PDF to the PR that adds/updates the PDFs and make the PDFs available on the website, too. Or do you think, this is a bad idea and we should add them manually from the artifacts @vsoch?

**Comment by Chilipp** [comment]
hmm, I see that the [pull-request-action](https://github.com/marketplace/actions/pull-request-action) does not work to open PRs on forks as you cannot specify the repository. So the strategy I proposed won't work. Then we add the PDFs manually to the PRs and we're done here

You really created a nice new tool here @vsoch =) Thanks a lot!

**Comment by vsoch** [comment]
Why do we need a fork? This is all under the same org (sorse) at this point. If the action is run on push to master to then check if there need to pdfs generated, it would work the same as the other actions we've worked on. Anyhoo, yes please take it from here.

**Comment by vsoch** [comment]
Also I rendered an example with a prettified date in my action repo, for comparison if needed https://github.com/rseng/pdf-generator/blob/master/paper/sorse.pdf.

**Comment by vsoch** [comment]
@Chilipp it should work fine if you run the action on merges to master, and then open a PR only if/when pdfs are generated.

**Comment by vsoch** [comment]
hmm so looks like we have some issues - the path of the folder _events is not really the path we want, and then the $pdf_generator_fileprefix$ is empty, and there is an extra / with programme/. I need to go for a quick run because I have to prepare for a meeting but let's discuss how we can fix this up. We need a solution that doesn't hard code something into the action but allows for generation of the url you need here.

**Comment by Chilipp** [comment]
I used `realpath` in https://github.com/rseng/pdf-generator/pull/4/commits/1b235e163dbfffbecc800c0b8478ece47c1702ad#. I am not in a rush as I am anyway currently reviewing a paper. But if you like, I can make some suggestions to https://github.com/rseng/pdf-generator/pull/5

**Comment by vsoch** [comment]
Yes I have an idea! I have a meeting now but will work on it soon :) Also, realpath won't fix the issue with _events not being the correct url. What I want to try is changing the workdir.

**Comment by Chilipp** [comment]
alright. I'll wait for it (it's anyway pretty late here and I have to finish this review, first)

**Comment by vsoch** [comment]
okay the file prefix I literally just forgot to add :) Now I'm going to test changing the workdir.

**Comment by vsoch** [comment]
@Chilipp I think this is good to go.

File: _events/latex.template.sorse [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+  \sffamily\small
+
+  $if(formatted_doi)${\bfseries DOI:} \href{https://doi.org/$formatted_doi$}{\color{linky}{$formatted_doi$}}$endif$
+  \vspace{2mm}
+
+  {\bfseries Software}
+  \begin{itemize}
+    \setlength\itemsep{0em}
+    \item \href{$url$}{\color{linky}{SORSE}} \ExternalLink
+    \item \href{$url$$baseurl$/events/$pdf_generator_outdir$/$pdf_generator_fileprefix$/}{\color{linky}{Event Website}} \ExternalLink
```
Comment (Chilipp) [inline _events/latex.template.sorse:327]:
```suggestion
    \item \href{$url$$baseurl$/$pdf_generator_outdir$/$pdf_generator_fileprefix$/}{\color{linky}{Event Website}} \ExternalLink
```

there is no `events` in the url, see https://496-267395254-gh.circle-artifacts.com/0/SORSE/programme/software_demos/event-001/index.html for instance.

**Comment by Chilipp** [comment]
Awesome! Thanks @vsoch :smiley:

**Comment by Chilipp** [comment]
test failure is a known issue (https://github.com/RSE-leaders/researchsoftware.org/issues/16), so I'll merge this

**Pull request merged**